main/lsp/wrapper.h

```diff
@@ -40,32 +53,89 @@ class LSPWrapper final {
         DocumentHighlight = 9,
     };
 
+    /**
+     * Raw constructor; use create* factory methods to construct the wrapper. Note: Constructor is unwieldy so we can
+     * make class fields `const`.
+     */
+    LSPWrapper(std::shared_ptr<options::Options> opts, std::shared_ptr<LSPLoop> lspLoop,
```

Comment:
I wanted to make this private, but then that disallows `make_unique`. Is there an alternative here? (Maybe I have to use `new` and move it into a `unique_ptr`?)

Reply:
> Maybe I have to use new and move it into a unique_ptr

yes, this is the alternative

Reply:
It looks like the abseil recommendation is a combination of `new` and `absl::WrapUnique` (c.f. [here](https://abseil.io/tips/134)) but this is outside the area of C++ where I trust my own knowledge; it might be worth asking others on the team if they have a preference.